geomstats/learning/em_expectation_maximization.py

```diff
@@ -0,0 +1,265 @@
+               'random' : will select random uniformally train point as
+                         initial centroids
+
+                #TODO: implement kmeans initialisation
+                'kmeans' : will apply Riemannian kmeans to deduce
+                variances and means that the EM will use initially
+
+        tol : float
+            Convergence factor. If the difference of mean distance
+            between two step is lower than tol
+
+        mean_method: basestring
+            Specify the method to compute the mean.
+
+        point_type: basestring
+            Specify whether to use vector or matrix representation
+
+        Returns
+        -------
+        self : object
+            Returns the instance itself.
+        """
+        self.n_gaussian = n_gaussian
+        self.riemannian_metric = riemannian_metric
+        self.initialisation_method = initialisation_method
+        self.tol = tol
+        self.mean_method = mean_method
+        self.point_type = point_type
+        self._dimension = None
+        self.mixture_coefficients = None
+        self.variances = None
+        self.means = None
+        self.normalization_factor = None
+
+    def update_posterior_probabilities(self, posterior_probabilities):
+        """Posterior probabilities update function."""
+        self.mixture_coefficients = gs.mean(posterior_probabilities, 0)
+
+    def update_means(self, data, posterior_probabilities,
+                     lr_means, tau_means, max_iter=DEFAULT_MAX_ITER):
+        """Means update function."""
+        n_gaussian = posterior_probabilities.shape[-1]
+
+        mean = FrechetMean(
+            metric=self.riemannian_metric,
+            method=self.mean_method,
+            lr=lr_means,
+            tau=tau_means,
+            max_iter=max_iter,
+            point_type=self.point_type)
+
+        data_expand = gs.expand_dims(data, 1)
+        data_expand = gs.repeat(data_expand, n_gaussian, axis=1)
+
+        mean.fit(data_expand, weights=posterior_probabilities)
+        self.means = gs.squeeze(mean.estimate_)
+
+    def update_variances(self, data, posterior_probabilities):
+        """Variances update function."""
+        n_data, n_gaussian = data.shape[0], self.means.shape[0]
+
+        data_expand = gs.expand_dims(data, 1)
+        data_expand = gs.repeat(data_expand, n_gaussian, axis=1)
+        means = gs.expand_dims(self.means, 0)
+        means = gs.repeat(means, n_data, axis=0)
+
+        weighted_dist_means_data = ((self.riemannian_metric.dist(
+            data_expand, means) ** 2) *
+            posterior_probabilities).sum(0) / \
+            posterior_probabilities.sum(0)
+
+        self.variances = \
+            self.normalization_factor.find_variance_from_index(
+                weighted_dist_means_data)
+
+    def _expectation(self, data):
+        """Update the posterior probabilities."""
+        probability_distribution_function = \
+            GaussianDistribution.gaussian_pdf(data,
```

Comment:
Cf my above comment: "Gaussian" is redundant here.

Reply:
and see comment on indentation: if you change the name, you will have to re-indent everything with this syntax.